platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -447,6 +451,10 @@ public boolean onMoveBegin(MoveGestureDetector detector) {
     public boolean onMove(MoveGestureDetector detector, float distanceX, float distanceY) {
       // first move event is often delivered with no displacement
       if (distanceX != 0 || distanceY != 0) {
+        if (annotationManager.onMove(distanceX, distanceY)) {
```

Comment:
You can skip `Projection#fromScreenLocation` conversion, and associated with it expensive JNI call, by passing coordinates returned by the `detector.getMoveObject(0)`.

Reply:
I haven't figured out how we can avoid `Projection#fromScreenLocation`. By using `detector.getMoveObject(0)` we can get the current screen coordinates which will avoid `Projection#toScreenLocation` call.